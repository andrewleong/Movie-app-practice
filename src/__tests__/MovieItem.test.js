import React from 'react';
import { shallow, mount } from 'enzyme';
import MovieItemComponent from '../components/MovieItem';


describe('(Component) MovieList', () => {

    const movie = {
        id: 1,
        title: "Movie 1",
        poster_path: '1234',
        release_date: '12-2-2019'
    }

    const wrapper = shallow(
        <MovieItemComponent
            movie={ movie }
        />
    );

    it('renders without crashing', () => {
        expect(wrapper).toHaveLength(1);
    });

    it('render movie item props correctly', () => {
        expect(wrapper.props()).toMatchSnapshot();
    });
});
