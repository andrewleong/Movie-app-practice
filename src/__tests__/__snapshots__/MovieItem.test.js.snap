// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) MovieList render movie item props correctly 1`] = `
Object {
  "alt": "",
  "children": Array [
    <img
      alt=""
      src="https://image.tmdb.org/t/p/w300/1234"
    />,
    <h5>
      Movie 1
    </h5>,
    <h6>
      12-2-2019
    </h6>,
  ],
  "className": "col-sm-12 movie-item-container",
  "onClick": undefined,
  "src": "https://image.tmdb.org/t/p/w300/1234",
}
`;
